app/src/newest/java/net/gini/android/gvlexample/GVLExampleApp.java

```diff
@@ -0,0 +1,35 @@
+package net.gini.android.gvlexample;
+
+import android.app.Application;
+import android.support.annotation.NonNull;
+
+import net.gini.android.vision.GiniVisionApplication;
+import net.gini.android.vision.network.GiniVisionNetwork;
+import net.gini.android.vision.network.GiniVisionNetworkHandler;
+
+/**
+ * Created by Alpar Szotyori on 27.11.2017.
+ *
+ * Copyright (c) 2017 Gini GmbH.
+ */
+
+public class GVLExampleApp extends Application implements GiniVisionApplication {
```

Comment:
Why is the `GiniVisionApplication` interface needed? I mean, you can also get the `GiniVisionNetwork` directly casting the `GVLExampleApp`, right? One thing that could make sense is using this interface for testing, but you are no injecting it in the `doSendFeedback` method.
Also I thought that all the network handling was done by the GVL itself, not by the new networking library. I was expecting the App sending the feedback to the GVL and then this one passing it to the new networking library, this way they only would have to access the GVL twice (one to start it and the other one to send the feedback).

Reply:
The `GiniVisionApplication` is the interface to which classes in GVL talk to to get an instance to the `GiniVisionNetwork` implementation. By using an interface we make it obvious what we need from the `Application` subclass. Casting without publishing an interface would be like praying: I pray that the Application has the `getGiniVisionNetwork()` method, and the prayer never be fulfilled 😄.

An important point of abstracting networking away behind the `GiniVisionNetwork` interface is to keep networking implementations outside of GVL. This means clients who use GVL with their own networking can do so in the future, too. Also unused GVL networking code won't be in their apps. Clients who don't want to deal with networking can use the default implementation and don't have to integrate the Gini API SDK themselves. This way we have flexibility and ease-of-use.

We can say, that the default implementation is logically part of GVL, but technically it's an independent package. The default implementation depends on the `GiniVisionNetwork` and related interfaces and classes.

Regarding feedback I find it useful to require networking implementations to implement it. Feedback is sent from views outside of GVL and very often we have issues with clients not implementing feedback right. At least for those who are fine with the default implementation we can make sure, that it's done right. The App sends the feedback to the implementation of `GiniVisionNetwork` - as stated above the default implementation of this is logically part of GVL - so we can say it sends it to the GVL. They do access the GVL twice: start it and send feedback.